test/test/assets/screenshots/review.html

```diff
@@ -128,127 +131,244 @@
+      }
+    }
+  }
 }
 
-// Remove the '?' at the front of the search string.
-const query = location.search.substr(1);
-if (!query) {
-  // Present a simple menu of screenshots to review.
-  h1.textContent = 'Select a screenshot or platform to review:';
+document.body.addEventListener('keydown', keyHandler);
+document.body.addEventListener('keyup', keyHandler);
+window.addEventListener('blur', focusHandler);
+window.addEventListener('focus', focusHandler);
 
-  const appendToList = (text) => {
-    const li = document.createElement('li');
-    document.body.appendChild(li);
 
-    const a = document.createElement('a');
-    a.textContent = text;
-    a.href = '?' + text;
-    li.appendChild(a);
-  };
+applyFilters();
 
-  for (const screenshot of allScreenshots()) {
-    appendToList(screenshot);
-  }
-  for (const platform of platforms) {
-    appendToList(platform);
-  }
-} else {
-  h1.textContent = 'Screenshots: ' + query;
-  h2.textContent = 'Mouse over a screenshot and use "shift" to see the new ' +
-      'version or "ctrl" to see changed areas in red.';
-
-  const keyHandler = (event) => {
-    if (event.shiftKey) {
-      document.body.classList.add('shift');
-    } else {
-      document.body.classList.remove('shift');
-      if (event.ctrlKey) {
-        document.body.classList.add('ctrl');
-      } else {
-        document.body.classList.remove('ctrl');
-      }
-    }
-  };
 
-  const focusHandler = () => {
-    // When we gain or lose focus, the keys that were pressed before are not
-    // necessarily pressed now.  Clear those states.
-    document.body.classList.remove('shift');
-    document.body.classList.remove('ctrl');
+function appendOption(parentElement, key, value, onChange) {
+  const container = document.createElement('div');
+  container.className = 'optionContainer';
+  parentElement.appendChild(container);
+
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+  input.dataset.key = key;
+  input.dataset.value = value;
+  container.appendChild(input);
+
+  const label = document.createElement('label');
+  label.textContent = value;
+  container.appendChild(label);
+
+  input.onchange = onChange;
+  allOptions[key] = allOptions[key] || {};
+  allOptions[key][value] = input;
+
+  // Make the label clickable as well.
+  label.onclick = () => {
+    input.checked = !input.checked;
+
+    // The onchange handler of the input not triggered by changing "checked" via
+    // the API.  So call the onChange handler explicitly.
+    onChange();
   };
 
-  document.body.addEventListener('keydown', keyHandler);
-  document.body.addEventListener('keyup', keyHandler);
-  window.addEventListener('blur', focusHandler);
-  window.addEventListener('focus', focusHandler);
+  return container;
+}
 
-  const appendScreenshot = (labelText, platform, screenshot) => {
-    const container = document.createElement('div');
-    container.classList.add('container');
+function appendScreenshot(platform, suite, test) {
```

Comment:
Hm. I'm not sure if I like that the `test` parameter for this method is not the same as the `test` variable in the calling context. Instead, it's equal to `prefix` + `-` + `test`.
I feel like either this should have a different name (`prefixedTest` maybe?), the test variable in the main loop should have a different name, or at least this should have some JSDoc explaining that this wants the test name with prefix included.

Reply:
Good point.  Fixed.